modules/external-api/src/main/java/org/opencastproject/external/endpoint/EventsEndpoint.java

```diff
@@ -312,11 +307,27 @@ public void setCommonEventCatalogUIAdapter(CommonEventCatalogUIAdapter eventCata
   )
   public void addCatalogUIAdapter(EventCatalogUIAdapter catalogUIAdapter) {
     catalogUIAdapters.add(catalogUIAdapter);
+    invalidateOrgCatalogUIAdaptersMapFor(catalogUIAdapter);
   }
```

Comment:
Instead of just invalidating the cache, you could add the adapter. What's the reason for doing it lazily?

Reply:
I find this mentally easier to reason about such a system: only one place where the cache is filled and one place where we invalidate the cache. In practice, this should also nearly never happen. Usually, the number of adapters is fixed, and these methods are called only during boot.

Reply:
Do you have to invalidate the cache on adding an adapter, wouldn't just adding the adapter be sufficient? Then you would still have only one place where the cache is filled and one place where it's invalidated.